pkg/scalers/azure_blob_scaler.go

```diff
@@ -143,13 +143,6 @@ func parseAzureBlobMetadata(config *ScalerConfig, logger logr.Logger) (*azure.Bl
 		config.PodIdentity = kedav1alpha1.AuthPodIdentity{Provider: kedav1alpha1.PodIdentityProviderAzure}
 	}
 
-	// FIXME: DEPRECATED to be removed in v2.12
-	if val, ok := config.TriggerMetadata["metricName"]; ok {
-		meta.MetricName = kedautil.NormalizeString(fmt.Sprintf("azure-blob-%s", val))
-	} else {
-		meta.MetricName = kedautil.NormalizeString(fmt.Sprintf("azure-blob-%s", meta.BlobContainerName))
-	}
-
```

Comment:
The feature removed is the `metricName` parameter on every scaler who doesn't use it as something internal of the scaler. If the scaler only uses it for renaming HPA metric, it should be removed, keeping only the case of unset, in this case:
```golang
meta.MetricName = kedautil.NormalizeString(fmt.Sprintf("azure-blob-%s", meta.BlobContainerName))
```
The metric name itself it's required because we need it for naming the metric in the HPA (and we still need it), but previously users could provide a custom name for the metric, which is the deprecated feature. The change is that after 2.12 users won't be able to provide a custom name and KEDA will generate a unique name internally (the code above for example)

Reply:
Yes, this is exactly what I thought. Thats why I pinged you on slack haha :laughing: . Thanks for the clarification!